va/tlsalpn.go

```diff
@@ -58,36 +60,46 @@ func certAltNames(cert *x509.Certificate) []string {
 
 func (va *ValidationAuthorityImpl) tryGetChallengeCert(
 	ctx context.Context,
-	identifier identifier.ACMEIdentifier,
-	tlsConfig *tls.Config,
+	ident identifier.ACMEIdentifier,
 ) (*x509.Certificate, *tls.ConnectionState, core.ValidationRecord, error) {
-
-	allAddrs, resolvers, err := va.getAddrs(ctx, identifier.Value)
 	validationRecord := core.ValidationRecord{
-		DnsName:           identifier.Value,
-		AddressesResolved: allAddrs,
-		Port:              strconv.Itoa(va.tlsPort),
-		ResolverAddrs:     resolvers,
+		DnsName: ident.Value,
+		Port:    strconv.Itoa(va.tlsPort),
 	}
-	if err != nil {
-		return nil, nil, validationRecord, err
+
+	var addrs []net.IP
+	switch ident.Type {
+	case identifier.TypeDNS:
+		// Resolve IP addresses for the identifier
+		dnsAddrs, dnsResolvers, err := va.getAddrs(ctx, ident.Value)
+		if err != nil {
+			return nil, nil, validationRecord, err
+		}
+		addrs, validationRecord.ResolverAddrs = dnsAddrs, dnsResolvers
+	case identifier.TypeIP:
+		addrs = []net.IP{net.ParseIP(ident.Value)}
+	default:
+		// This should never happen. The calling function should check the
+		// identifier type.
+		return nil, nil, validationRecord, fmt.Errorf("unknown identifier type: %s", ident.Type)
 	}
+	validationRecord.AddressesResolved = addrs
```

Comment:
We should only store `AddressesResolved` in the `TypeDNS` case. In the IP case, the address wasn't resolved; and more to the point, it's redundant. The address is already present in the identifier (which is part of the authz object).